prime-router/src/main/kotlin/azure/DatabaseAccess.kt

```diff
@@ -137,6 +137,26 @@ class DatabaseAccess(private val create: DSLContext) : Logging {
             .into(Task::class.java)
     }
 
+    /**
+     * Returns true if there is already a record in the report_file table that matches the passed in [senderName],
+     * [senderOrgName], and [digest]
+     */
+    fun checkForDuplicate(
+        senderName: String,
+        senderOrgName: String,
+        digest: ByteArray,
+        txn: DataAccessTransaction? = null
+    ): Boolean {
+        val ctx = if (txn != null) DSL.using(txn) else create
+        return ctx
+            .fetchExists(
+                ctx.selectFrom(REPORT_FILE)
+                    .where(REPORT_FILE.SENDING_ORG.eq(senderOrgName))
+                    .and(REPORT_FILE.SENDING_ORG_CLIENT.eq(senderName))
+                    .and(REPORT_FILE.BLOB_DIGEST.eq(digest))
```

Comment:
Are some indexes wanted / needed for this query to make it scalable?

Reply:
We will definitely need an index to make this speedy.

Reply:
Please note the 'outstanding work needed' comment in the PR - yes, indexes may be needed. Working with Chris on it.

Reply:
Added index and flyway script